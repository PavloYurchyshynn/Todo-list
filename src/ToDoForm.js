import { useState } from 'react'

function ToDoForm({ addTask }) {
    const [userInput, setUserInput] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault()
        addTask(userInput)
        setUserInput('')
    }

    const handleChange = (e) => {
        setUserInput(e.currentTarget.value)
    }

    const handleKeyPress = (e) => {
        if (e.key === 'Enter') {
            handleSubmit(e)
        }
    }

    return (
        <form data-testid='todoform-1' onSubmit={handleSubmit}>
            <input
                value={userInput}
                type='text'
                onChange={handleChange}
                onKeyDown={handleKeyPress}
                placeholder='Введіть значення'
            />
            <button>Зберегти</button>
        </form>
    )
}

export default ToDoForm