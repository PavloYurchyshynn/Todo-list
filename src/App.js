import { useState } from 'react'
import ToDo from './ToDo'
import ToDoForm from './ToDoForm'


function App() {

  const [todos, setTodos] = useState([])

  const addTask = () => {

  }

  const removeTask = () => {

  }

  const handleToggle = () => {

  }

  return (
    <div className="App">
      <header>
        <h1>Список завдань: {todos.length}</h1>
      </header>
      <ToDoForm />
      {todos.map((todo) => {
        return (
          <ToDo key={todo.id}  />
        )
      })}
    </div>
  );
}

export default App;
