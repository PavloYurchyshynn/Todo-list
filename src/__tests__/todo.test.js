import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import App from '../App'
import ToDo from '../ToDo'
import ToDoForm from '../ToDoForm'

afterEach(() => {
    cleanup()
})

test('should render todo form component', () => {
    render(<ToDoForm />)
    const todoFormElement = screen.getByTestId('todoform-1')
    expect(todoFormElement).toBeInTheDocument();
})

test('should render app component', () => {
    render(<App />)
    const appElement = screen.getByTestId('app-1')
    expect(appElement).toBeInTheDocument();
    expect(appElement).toHaveTextContent('Список завдань:');
})

test('should render todo component', () => {
    const todo = { id: 1, task: 'text', complete: true }
    render(<ToDo todo={todo}/>)
    const todoElement = screen.getByTestId('todo-1')
    expect(todoElement).toBeInTheDocument()
})

test('matches snapshot', () => {
    const todo = { id: 1, task: 'text', complete: true }
    const tree = renderer.create(<ToDo todo={todo}/>).toJSON()
    expect(tree).toMatchSnapshot()
})