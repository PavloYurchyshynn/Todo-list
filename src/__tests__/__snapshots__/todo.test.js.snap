// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="item-todo"
  data-testid="todo-1"
>
  <div
    className="item-text strike"
    onClick={[Function]}
  >
    text
  </div>
  <div
    className="item-delete"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
